### ISSUE TITLE
Online storage of configuration folder

### ISSUE THREAD

**Issue by manparvesh** [dude-pa/dude#8]
Feature description:
-------------------
User should be able to save their config somewhere online so that they can sync their data over different devices.

Any ideas on how to implement it?
---------------------------------
Possible solution can be creating a github repository and saving the data to it after the user enters a command.  
I was able to initialize a repo inside the config file, stage the changes and commit to the file using [dulvich](https://github.com/jelmer/dulwich), but I was not able to connect it to github. I tried [PyGithub](https://github.com/PyGithub/PyGithub), but no luck.

**Comment by thee-engineer** [comment]
How about using GoogleDrive/iCloud or other cloud storage services? Give the user a choice.

**Comment by manparvesh** [comment]
@thee-engineer yes, this is just an initial idea. It can obviously be expanded!

**Comment by nabarunchatterjee** [comment]
IMHO the easiest to implement would be AWS s3 for storing configs. For git storage we can ask the user for github repo url and the rest can be taken care of by
 [gitpython](http://gitpython.readthedocs.io/en/stable/)

**Comment by manparvesh** [comment]
@nabarunchatterjee we need the configs to be accessible only by the user, so I think it would be better if we found a way to save in their own account; hence the idea of using git repo as a storage.
I don't know much about `gitpython`, so if you're interested to work on it, I'd be happy to receive an MR!

**Comment by sarathsoma** [comment]
@manparvesh IMHO we can use [Drive API](https://developers.google.com/drive/api/v3/appdata) to store the appdata privately in the user's google drive and use oauth to sync the settings across devices. Could be a good starting point for using yoda on android devices ( if someone hacks in to develop an app on top of yoda!)

**Comment by manparvesh** [comment]
@sarathsoma that seems like a good idea!

**Comment by thee-engineer** [comment]
GitHub Gists would also be a possible solution or simple "pastebin" style platforms. There are plenty of options for cloud storage these days and especially for config storage. The more implementations and the more options the user has, the better.
